src/Dialect/Mlir/DialectBuilder.hpp

```diff
@@ -259,6 +259,84 @@ struct GenericAffineBuilder final : DialectBuilder {
 using AffineBuilder =
     GenericAffineBuilder<mlir::AffineLoadOp, mlir::AffineStoreOp>;
 
+//===----------------------------------------------------------------------===//
+// LLVM Builder
+//===----------------------------------------------------------------------===//
+
+struct LLVMBuilder final : DialectBuilder {
+  LLVMBuilder(mlir::OpBuilder &b, mlir::Location loc)
+      : DialectBuilder(b, loc) {}
+  LLVMBuilder(DialectBuilder &db) : DialectBuilder(db) {}
+
+  // AddressOfOp
+  mlir::Value addressOf(mlir::LLVM::GlobalOp op) const;
+
+  // AllocaOp
+  mlir::Value _alloca(
+      mlir::Type resultType, mlir::Value size, int64_t alignment) const;
+
+  // BitcastOp
+  mlir::Value bitcast(mlir::Type type, mlir::Value val) const;
+  mlir::Value bitcastI8Ptr(mlir::Value val) const;
+  mlir::Value bitcastI8PtrPtr(mlir::Value val) const;
+
+  // CallOp
+  mlir::Value call(mlir::ArrayRef<mlir::Type> resultTypes,
+      llvm::StringRef funcName, mlir::ArrayRef<mlir::Value> inputs) const;
+  mlir::Value call(mlir::ArrayRef<mlir::Type> resultTypes,
+      mlir::FlatSymbolRefAttr funcSymbol,
+      mlir::ArrayRef<mlir::Value> inputs) const;
+
+  // ConstantOp
```

Comment:
Would it make sense to have 2 versions, one with int and one with double val?

Reply:
Yes, I will make two versions.